### DISCUSSION TITLE
anyio.BrokenResourceError

### DISCUSSION THREAD

**Discussion by haf** [encode/starlette#1622] [Potential Issue] [answered]
We're having trouble with `anyio.WouldBlock` and `anyio.BrokenResourceError` crashing our FastAPI app. This stems from our usage of middleware, which in turn is a requirement for us to keep the code manageable.

I would ask that you read through [this thread](https://github.com/tiangolo/fastapi/issues/4041#issuecomment-1115820813); possibly even backwards from the bottom up, as the bottom has the latest repros and stacktraces.

For me [here's my stacktrace](https://github.com/tiangolo/fastapi/issues/4041#issuecomment-1115820813) — and it stems from a `await next(req)` call without anything else.

**Comment by Kludex** [comment]
Does this branch solves the issue for you? https://github.com/encode/starlette/pull/1441

**Reply by haf** [reply]
Perhaps you can release this as a fixed version of a package and I can upgrade and try it?

**Reply by Kludex** [reply]
No... I can't. That's not how we work. You can already use that branch for testing purposes.

You'll probably want to create a minimal example of how you're getting your issue, and then use that branch.

**Comment by haf** [comment]
Hi @Kludex I don't know. Is there a package I can reference with this code?

**Reply by Kludex** [reply]
You can turn your FastAPI code into Starlette, and then install that branch. Check https://stackoverflow.com/questions/20101834/pip-install-from-git-repo-branch for more information.

**Reply by haf** [reply]
Do you mean turn all middleware into Starlette-binding middleware, including what I take from packages?

**Comment by Kludex** [comment, marked as answer]
This was already solved on the latest Starlette.

I believe it was solved on Starlette 0.21.0. :eyes:
